src/lib/pages/account-details/data.ts

```diff
@@ -152,3 +201,251 @@ export const useUserAssetInfos = (
+        token,
+        {
+          logo: assetInfo.logo,
+          precision: assetInfo.precision,
+          value: calculateAssetValue(
+            toToken(token.amount, assetInfo.precision),
+            assetInfo.price as USD<number>
+          ),
+        }
+      )
+    : token;
+};
+
+const addTotal = (
+  oldTotal: Option<TokenWithValue>,
+  token: TokenWithValue
+): TokenWithValue =>
+  !oldTotal
+    ? token
+    : {
+        ...oldTotal,
+        amount: oldTotal.amount.add(token.amount) as U<Token<Big>>,
+        value: oldTotal.value?.add(token.value ?? 0) as USD<Big>,
+      };
+
+const calBonded = (
+  totalDelegations: Option<Record<string, TokenWithValue>>,
+  totalUnbondings: Option<Record<string, TokenWithValue>>
+) => {
+  if (!totalDelegations || !totalUnbondings) return undefined;
+
+  return Object.keys(totalDelegations).reduce(
+    (total, denom) => ({
+      ...total,
+      [denom]: addTotal(totalUnbondings[denom], totalDelegations[denom]),
+    }),
+    {} as Record<string, TokenWithValue>
+  );
+};
+
+export const useUserDelegationInfos = (walletAddress: HumanAddr) => {
+  const { data: rawStakingParams, isLoading: isLoadingRawStakingParams } =
+    useStakingParams();
+  const { data: assetInfos, isLoading: isLoadingAssetInfos } = useAssetInfos();
+  const { data: validators, isLoading: isLoadingValidators } = useValidators();
+
+  const { data: rawDelegations, isLoading: isLoadingRawDelegations } =
+    useDelegations(walletAddress);
+  const { data: rawUnbondings, isLoading: isLoadingRawUnbondings } =
+    useUnbondings(walletAddress);
+  const { data: rawRewards, isLoading: isLoadingRawRewards } =
+    useDelegationRewards(walletAddress);
+  const { data: rawRedelegations, isLoading: isLoadingRawRedelegations } =
+    useRedelegations(walletAddress);
+  const { data: rawCommission, isLoading: isLoadingRawCommission } =
+    useCommission(addrToValoper(walletAddress));
+
+  const isLoading =
+    isLoadingRawStakingParams || isLoadingAssetInfos || isLoadingValidators;
+  const data: UserDelegationsData = {
+    stakingParams: undefined,
+    isValidator: undefined,
+    isLoading,
+    totalBonded: undefined,
+    isLoadingTotalBonded:
+      isLoading || isLoadingRawDelegations || isLoadingRawUnbondings,
+    totalDelegations: undefined,
+    delegations: undefined,
+    isLoadingDelegations: isLoading || isLoadingRawDelegations,
+    totalUnbondings: undefined,
+    unbondings: undefined,
+    isLoadingUnbondings: isLoading || isLoadingRawUnbondings,
+    totalRewards: undefined,
+    rewards: undefined,
+    isLoadingRewards: isLoading || isLoadingRawRewards,
+    redelegations: undefined,
+    isLoadingRedelegations: isLoading || isLoadingRawRedelegations,
+    totalCommission: undefined,
+    isLoadingTotalCommission: isLoading || isLoadingRawCommission,
+  };
+
+  if (rawStakingParams && assetInfos && validators) {
+    const stakingParams = {
+      ...rawStakingParams,
```

Comment:
No need `...rawStakingParams` type StakingParams only have logo and precision

Reply:
`StakingParams` extends `RawStakingParams`